autoload/cmake/switch.vim

```diff
@@ -72,14 +72,32 @@ function! cmake#switch#GetCurrent() abort
     return s:config
 endfunction
 
+" Get path to parent build configuration, possibly reduced relatively to CWD.
+"
+" Returns:
+"     String
+"         (unescaped) path to parent build configuration
+"
+function! cmake#switch#GetBuildDir() abort
```

Comment:
This doesn't work in all cases unfortunately. When running CMake from e.g. a subdirectory of the project root (which Vim-CMake supports as of 743e3d316db3d3f08e348973c4e9c6807659c954), `build_dir` will be relative to the project root, and not to the CWD.

I would suggest to compute `l:build_dir` (which we should probably rename to `l:build_dir_parent`) the same as we compute `s:source_dir` in autoload/cmake.vim. When the build dir parent is not below the current dir, it will expand to a full path, not a relative path, but It doesn't matter in my opinion.

Let's also rename the function to `GetBuildDirParent()`, in light of the other comment I left in plugin/cmake.vim, and let's make this function "private", as we only have to access it from within this file.